src/extension.ts

```diff
@@ -153,16 +157,23 @@ export function activate(context: vscode.ExtensionContext): void {
         if (target instanceof vscode.Uri) {
             fsPath = target.fsPath;
         } else {
-            fsPath = await util.showWorkspaceFoldersQuickPick("Select the folder to deploy");
+            fsPath = await util.showWorkspaceFoldersQuickPick("Select the folder to deploy", this.properties);
             node = target;
         }
 
         if (!node) {
-            node = <IAzureNode<WebAppTreeItem>>await tree.showNodePicker(WebAppTreeItem.contextValue);
+            try {
+                node = <IAzureNode<WebAppTreeItem>>await tree.showNodePicker(WebAppTreeItem.contextValue);
+            } catch (err2) {
+                if (err2 instanceof UserCancelledError) {
```

Comment:
Do `parseError(err2).isUserCancelledError` (we've had weird bugs in the past where `instanceof` doesn't work if there are somehow multiple versions of the ui package)